Upgrade from `emoji-button` to `PicMo`

paescuj opened the pull request:
## Description

The package `@joeattardi/emoji-button` which is used as the emoji picker has been deprecated a while ago. The successor of it is the `picmo` package:

> Emoji Button is now PicMo! Please install the 'picmo' package, see https://picmojs.com for details

### Design

The design has changed quite a bit. I've tried to adapt it as much as possible to the one of Directus.
Please let me know if anything needs to be adjusted. For example, I'm not so sure about the new category bar...

<table>
<thead><tr>
<th></th>
<th>Light</th>
<th>Dark</th>
</tr></thead>
<tbody><tr>
<th>Before</th>
<td><img width="350" src="https://user-images.githubusercontent.com/5363448/203474270-a4a46311-3eaa-4b25-a55b-eed0face0670.png"></td>
<td><img width="350" src="https://user-images.githubusercontent.com/5363448/203474278-f7f43647-6681-411d-9ee7-84ae8ff9a513.png"></td>
</tr><tr>
<th>After</th>
<td><img width="350" src="https://user-images.githubusercontent.com/5363448/203474354-ad94d84a-e58d-40b5-807d-b3a17c96ca70.png"></td>
<td><img width="350" src="https://user-images.githubusercontent.com/5363448/203474359-50c092a6-546b-4284-9d7f-3bdd8523097f.png"></td>
</tr></tbody>
</table>

<table>
<thead><tr>
<th></th>
<th><code>picmo</code></th>
<th><code>@joeattardi/emoji-button</code></th>
</tr></thead>
<tbody><tr>
<td align="center"><b>Default Theme</b><br>(for reference)</td>
<td><img width="350" src="https://user-images.githubusercontent.com/5363448/203575195-2dfc9483-e81b-4389-a878-5f321843f944.png"></td>
<td><img width="350" src="https://user-images.githubusercontent.com/5363448/203575205-11784bb7-5c20-4ba6-8212-17d41295bfac.png"></td>
</tr></tbody>
</table>

### Implementation

The emojis itself are now internationalized and I've added the strings from the picker UI to the translation file. Currently, a workaround is required for the "recently used" category to be properly translated. If I get the time, I'll report this in `picmo`s repo.

Note: Since the emoji data needs to be loaded dynamically, the component is now async. How do you usually deal with this? Is the use of `<Suspense>` acceptable?

Related to #12119.

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

paescuj committed:
Upgrade from 'emoji-button' to 'picmo'

paescuj committed:
Remove @joeattardi/emoji-button

paescuj committed:
Use clearer translation key

paescuj committed:
Move styles near component

Nitwel reviewed (changes requested):
Awesome that you spend the time updating this ❤️!

There just a few minor question / things that need to be covered before we can merge this.

There is a small border around the icons that gets hidden on hover but I would say we can get rid of that at all. 
![grafik](https://user-images.githubusercontent.com/22577866/206754104-3ac77d72-1ca6-4fe7-835b-f4d19d6edc1f.png)

~~And another thing I noticed is that you can't select the color of an emoji currently and I think we had that before as an option.~~ I just clicked on emojis that didn't had the feature...

paescuj committed:
Fix border radius on emojis

paescuj commented:
> Awesome that you spend the time updating this ❤️!

Thank you ❤️ I really appreciate your review on this!

> There just a few minor question / things that need to be covered before we can merge this.
> 
> There is a small border around the icons that gets hidden on hover but I would say we can get rid of that at all. ![grafik](https://user-images.githubusercontent.com/22577866/206754104-3ac77d72-1ca6-4fe7-835b-f4d19d6edc1f.png)

Wow, good catch! 😃 The border radius from the default theme was far too high. I decided to keep a little radius because IMO it makes things look a little smoother. With the property set directly on the emoji it now also applies to `:focus` 👍 

> ~And another thing I noticed is that you can't select the color of an emoji currently and I think we had that before as an option.~ I just clicked on emojis that didn't had the feature...

Yes, it's a bit hidden 🙂

paescuj commented on app/.storybook/preview.js:
```diff
@@ -16,7 +16,7 @@ const router = createRouter({
 });
 
-const i18n = createI18n();
+const i18n = createI18n({ legacy: false });
```
This was required for the emoji picker to be rendered in Storybook. Without this setting the following error was shown:
<img width="937" alt="Screenshot 2022-12-13 at 19 03 18" src="https://user-images.githubusercontent.com/5363448/207410588-38177582-9a91-475c-8acf-433ee539b92f.png">

I assume this is somehow related to the fact that the component is async. Since this option "only" applies to Storybook, I thought it to be acceptable and haven't looked at it more closely.

paescuj commented:
> Another thing I just noticed is that PicMo is reaching out for `https://cdn.jsdelivr.net/npm/emojibase-data@latest/en/data.json` and caching it in the browser, though I would prefer it to load that from the local `node_modules` instead. I'm not sure if that is an option with PicMo or if it is even better to fetch it from a cdn as it then will be always up to date with the latest emoji standard.

Strange! I actually configured it to load the data locally. I'm going to look into it!

paescuj committed:
Finalize emoji picker

- Fix path to emoji messages
- Add some comments to make it clearer
- Close when position is lost (e.g. route change)

paescuj commented:
> Another thing I just noticed is that PicMo is reaching out for `https://cdn.jsdelivr.net/npm/emojibase-data@latest/en/data.json` and caching it in the browser, though I would prefer it to load that from the local `node_modules` instead. I'm not sure if that is an option with PicMo or if it is even better to fetch it from a cdn as it then will be always up to date with the latest emoji standard.

Another good catch from your side 😄🙈 Turns out I forgot to update the path to the emoji messages file when moving the component in a sub-folder. This is now fixed 👍 I've also made the following additions:
- Add some comments the code to make it clearer
- Add warnings for the unexpected case that data cannot be loaded
- Close the picker when position is lost (e.g. on route change) - previously the picker was briefly displayed in the wrong place

I think we should be good to go now 😃 

> [...] though I would prefer it to load that from the local node_modules instead. I'm not sure if that is an option with PicMo or if it is even better to fetch it from a cdn as it then will be always up to date with the latest emoji standard.

I fully agree and now it works exactly like this! For the unexpected case that the data cannot be loaded locally, PicMo will attempt to fetch it from the CDN. Since the emoji data is stored in a separate module, it shouldn't be a problem to keep it up to date.

br41nslug committed:
Merge branch 'main' into emoji-button-to-picmo

br41nslug reviewed (commented):
`<Suspense>` Seems to be working fine in my local tests however this disclaimer on the VueJS site makes me doubt whether we should/can really rely on it yet 🤔 
> `<Suspense>` is an experimental feature. It is not guaranteed to reach stable status and the API may change before it does.

https://vuejs.org/guide/built-ins/suspense.html

rijkvanzanten committed:
Merge branch 'main' into emoji-button-to-picmo

rijkvanzanten commented:
![CleanShot 2022-12-23 at 14 20 50@2x](https://user-images.githubusercontent.com/9141017/209396781-120f6b57-1ac9-44c0-9d3f-397ceeb19c3b.png)

Some things feel a little "off":

1. Input is missing it's focus "glow" like we're using elsewhere
2. Is this using a different border radius and shadow? Lets match `v-menu` as closely as possible
3. Lets add an 'active' state and arrow-nub on this 
4. Any chance we can use material icons here? These icons are off-brand currently 🤔

rijkvanzanten commented:
Closing this for now to be revisited later
